features/lorawan/system/lorawan_data_structures.h

```diff
@@ -69,6 +69,43 @@ typedef uint32_t lorawan_time_t;
 
 #define LORAWAN_DEFAULT_QOS                         1
 
+#define BIT_SET_TEST(mask, bit)    (mask & (1 << bit))
+#define SET_BIT(mask, bit)         (mask |= (1 << bit))
+#define UNSET_BIT(mask, bit)       (mask &= ~(1 << bit))
+#define COUNT_SET_BITS(mask, output) \
+output = 0; \
+while (mask) { \
+    mask &= (mask - 1); \
+    output++; \
+} \
+
+
+inline uint8_t *write_four_bytes(uint32_t value, uint8_t *ptr)
+{
+    if (!ptr) {
```

Comment:
No implementation in header files

Reply:
It's an inline function. The problem is, it's harder to find a central location which accessible by every component. The idea was to make it available as a global helper without limiting its scope. Maybe we need to rethink.  If I remember correctly, there are other examples of inline function implementations in the headers in Mbed OS.

Reply:
Yes there are, but it does not mean it is a correct way. We should also clean those going forward.
You will need something like common_helpers.cpp to put implementation into.